#define _XOPEN_SOURCE / _POSIX_C_SOURCE twice defined in python and in /usr/include/features.h

### Problem

Currently, compilation of python bindings fails for me due to two macros being redefined.

**Expected behavior:** 

Compile to finish.

**Actual behavior:** 

In file included from /usr/include/python2.7/Python.h:8:0,
                 from ../ext/pybind11/include/pybind11/detail/common.h:111,
                 from ../ext/pybind11/include/pybind11/pytypes.h:12,
                 from ../ext/pybind11/include/pybind11/cast.h:13,
                 from ../ext/pybind11/include/pybind11/attr.h:13,
                 from ../ext/pybind11/include/pybind11/pybind11.h:43,
                 from ../src/python/py_oiio.h:48,
                 from ../src/python/py_imageinput.cpp:31:
/usr/include/python2.7/pyconfig.h:1161:0: error: "_POSIX_C_SOURCE" redefined [-Werror]
 #define _POSIX_C_SOURCE 200112L
 ^
In file included from /usr/include/x86_64-linux-gnu/c++/4.9/bits/os_defines.h:39:0,
                 from /usr/include/x86_64-linux-gnu/c++/4.9/bits/c++config.h:430,
                 from /usr/include/c++/4.9/bits/stl_algobase.h:59,
                 from /usr/include/c++/4.9/memory:62,
                 from ../src/python/py_oiio.h:34,
                 from ../src/python/py_imageinput.cpp:31:
/usr/include/features.h:164:0: note: this is the location of the previous definition
 #define _POSIX_C_SOURCE 200809L
 ^
In file included from /usr/include/python2.7/Python.h:8:0,
                 from ../ext/pybind11/include/pybind11/detail/common.h:111,
                 from ../ext/pybind11/include/pybind11/pytypes.h:12,
                 from ../ext/pybind11/include/pybind11/cast.h:13,
                 from ../ext/pybind11/include/pybind11/attr.h:13,
                 from ../ext/pybind11/include/pybind11/pybind11.h:43,
                 from ../src/python/py_oiio.h:48,
                 from ../src/python/py_imageinput.cpp:31:
/usr/include/python2.7/pyconfig.h:1183:0: error: "_XOPEN_SOURCE" redefined [-Werror]
 #define _XOPEN_SOURCE 600
 ^
In file included from /usr/include/x86_64-linux-gnu/c++/4.9/bits/os_defines.h:39:0,
                 from /usr/include/x86_64-linux-gnu/c++/4.9/bits/c++config.h:430,
                 from /usr/include/c++/4.9/bits/stl_algobase.h:59,
                 from /usr/include/c++/4.9/memory:62,
                 from ../src/python/py_oiio.h:34,
                 from ../src/python/py_imageinput.cpp:31:
/usr/include/features.h:166:0: note: this is the location of the previous definition
 #define _XOPEN_SOURCE 700
 ^
cc1plus: error: unrecognized command line option "-Wno-macro-redefined" [-Werror]
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.

### Versions

* OIIO branch/version: master (or 1.9 dev2)
* OS: Linux
* C++ compiler: gcc-4.9.4

--- a/src/python/py_imagebuf.cpp
+++ b/src/python/py_imagebuf.cpp
@@ -28,9 +28,10 @@
   (This is the Modified BSD License)
 */
 
+#include "py_oiio.h"
+
 #include <memory>
 
-#include "py_oiio.h"
 #include <OpenImageIO/platform.h>
 
 

--- a/src/python/py_imagebufalgo.cpp
+++ b/src/python/py_imagebufalgo.cpp
@@ -27,9 +27,10 @@
 
   (This is the Modified BSD License)
 */
+
+#include "py_oiio.h"
 #include <OpenImageIO/color.h>
 #include <OpenImageIO/imagebufalgo.h>
-#include "py_oiio.h"
 
 
 namespace PyOpenImageIO

--- a/src/python/py_oiio.h
+++ b/src/python/py_oiio.h
@@ -31,6 +31,9 @@
 #ifndef PYOPENIMAGEIO_PY_OIIO_H
 #define PYOPENIMAGEIO_PY_OIIO_H
 
+// Must include Python.h first to avoid certain warnings
+#include <Python.h>
+
 #include <memory>
 
 // Avoid a compiler warning from a duplication in tiffconf.h/pyconfig.h

